docs/tutorials/07_vpp-connection.md

```diff
@@ -0,0 +1,360 @@
+# Tutorial: Connect the plugin to the VPP
+
+This tutorial will illustrate how to utilize the [GoVPPMux plugin][1] in our hello world plugin to connect to the VPP and how to perform a synchronous, asynchronous and multi-request call in order to put configuration items.
+
+Requirements:
+* Complete and understand the ['Hello World Agent'](https://ligato.io/cn-infra/tutorials/01_hello-world) tutorial
+* Complete and understand the ['Plugin Dependencies'](https://ligato.io/cn-infra/tutorials/02_plugin-deps) tutorial
+* Installed VPP, or VPP binary file of the supported version
+
+The main task of this tutorial is to show how to use GoVPPMux to create VPP connection (southbound) and use various procedures to put or read its data. Because the main intent is to show how to work with the vpp, we do not use any northbound database to keep the example as simple as possible. 
+
+Our first step is to create a reliable connection with the VPP. THe vpp-agent uses [GoVPP][2] as VPP adapter. We add the GoVPPMux plugin (the GoVPP wrapper) to our hello world example:
+```go
+type HelloWorld struct {
+	GoVPPMux govppmux.API
+}
+```   
+
+And initialize it in the `main` right after the hello world plugin struct definition:
+```go
+func main() {
+	p := new(HelloWorld)
+	p.GoVPPMux = &govppmux.DefaultPlugin    // initialize with the default plugin instance
+	
+...
+	
+}
+``` 
+
+The GoVPPMux API allows other plugins to create their own API channels communicating with the VPP using GoVPP core. The channel can be either buffered or unbuffered and is used to pass requests and replies between the VPP and our plugin. Let's create one in the `Init`. At first, we need a variable of `Channe` type to have the VPP channel available for all hello world methods:
+```go
+type HelloWorld struct {
+	vppChan api.Channel
+
+	GoVPPMux govppmux.API
+}
+```
+
+Then initialize new VPP unbuffered channel:
+```go
+func (p *HelloWorld) Init() (err error) {
+	log.Println("Hello World!")
+
+	if p.vppChan, err = p.GoVPPMux.NewAPIChannel(); err != nil {
+		panic(err)
+	}
+	return nil
+}
+```
+
+Later, we use the channel to send or receive data, but at first, we need to prepare the data and send them in the way VPP can understand.
+
+#### 1. Binary API
+
+Note: this step is only explanatory and can be skipped
+
+The GoVPP is in general a toolset for the VPP management, providing high-level API for communication with GoVPP core and sending and receiving messages to/from the VPP via adapter - a component between GoVPP and the VPP, responsible for sending and receiving binary-encoded data via the VPP socket client (by default, but the VPP shared memory is also available when needed). It also provides a bindings generator for VPP JSON binary API definitions - the **binapi-generator**.
```

Comment:
This section should probably be part of GoVPP documentation and we could just add link here.